source/slang/slang-emit-cpp.cpp

```diff
@@ -2589,34 +2589,43 @@ void CPPSourceEmitter::emitModuleImpl(IRModule* module)
                             default:
 
                                 auto targetProgram = program->getTargetProgram(targetRequest);
-                                DiagnosticSink sink(linkage->getSourceManager());
+                                //DiagnosticSink sink(linkage->getSourceManager());
                                 CompileResult result =
-                                    targetProgram->getOrCreateEntryPointResult(index, &sink);
+                                    targetProgram->getOrCreateEntryPointResult(index, sink);
 
                                 Slang::ComPtr<ISlangBlob> blob;
                                 if (SLANG_FAILED(result.getBlob(blob)))
                                 {
-                                    SLANG_UNEXPECTED("No blob to emit");
-                                    return;
+                                    StringBuilder builder;
```

Comment:
The builder here doesn't look like it's used. It *might* be more convenient to build up your result using 

```
builder << "size_t __" << entryPointName << "Size = 0;\n"; 
```

etc, and just write out to writer at the end.